src/tribler-core/tribler_core/upgrade/db8_to_db10.py

```diff
@@ -0,0 +1,116 @@
+        self.old_db_path = old_db_path
+        self.new_db_path = new_db_path
+        self.shutting_down = False
+
+    async def update_convert_progress(self, amount, total, elapsed, message=""):
+        if self.notifier_callback:
+            elapsed = 0.0001 if elapsed == 0.0 else elapsed
+            amount = amount or 1
+            est_speed = amount / elapsed
+            eta = str(datetime.timedelta(seconds=int((total - amount) / est_speed)))
+            self.notifier_callback(
+                f"{message}\nConverted: {amount}/{total} ({(amount * 100) // total}%).\nTime remaining: {eta}"
+            )
+            await sleep(0.001)
+
+    async def convert_async(self, convert_command, total_to_convert, offset=0, message=""):
+        """
+        This method copies entries from one Pony db into another one splitting the process into chunks dynamically.
+        Chunks splitting uses congestion-control-like algorithm. Awaits are necessary so the
+        reactor can get an opportunity at serving other tasks, such as sending progress notifications to
+        the GUI through the REST API.
+        """
+        start_time = datetime.datetime.utcnow()
+        batch_size = 100
+
+        reference_timedelta = datetime.timedelta(milliseconds=1000)
+        while offset < total_to_convert:
+            if self.shutting_down:
+                break
+            end = offset + batch_size
+
+            batch_start_time = datetime.datetime.now()
+            convert_command(offset, batch_size)
+            batch_end_time = datetime.datetime.now() - batch_start_time
+
+            elapsed = (datetime.datetime.utcnow() - start_time).total_seconds()
+            await self.update_convert_progress(offset, total_to_convert, elapsed, message)
+            target_coeff = (batch_end_time.total_seconds() / reference_timedelta.total_seconds())
+            if target_coeff < 0.8:
+                batch_size += batch_size
+            elif target_coeff > 1.1:
+                batch_size = int(float(batch_size) / target_coeff)
+            # we want to guarantee that at least some entries will go through
+            batch_size = batch_size if batch_size > 10 else 10
+            self._logger.info("Converted: %i/%i %f ",
+                              offset + batch_size, total_to_convert, float(batch_end_time.total_seconds()))
+            offset = end
+
+    def get_table_entries_count(self, cursor, table_name):
+        return cursor.execute(f"SELECT COUNT(*) FROM {table_name};").fetchone()[0]
+
+    async def convert_table(self, cursor, table_name, column_names):
+        def convert_command(offset, batch_size):
+            sql_command = None
+            try:
+                sql_command = f"INSERT OR IGNORE INTO {table_name} ({column_names}) " + \
+                              f"SELECT {column_names} FROM old_db.{table_name} " + \
+                              f"LIMIT {batch_size} {('OFFSET ' + str(offset)) if offset else ''} ;"
```

Comment:
I'd add `ORDER BY  ROWID` just to be sure that SELECT returns rows in a stable order